src/main/java/edu/cornell/kfs/gl/batch/service/impl/CuPosterServiceImpl.java

```diff
@@ -17,4 +65,284 @@ protected KualiDecimal getPercentage(KualiDecimal amount, BigDecimal percent) {
         BigDecimal result = amount.bigDecimalValue().multiply(percent).divide(BDONEHUNDRED, 2, BigDecimal.ROUND_HALF_UP);
         return new KualiDecimal(result);
     }
+
+    /**
+     * Overridden to check for closed or expired accounts on ICR transactions, and to update
+     * such transactions to use continuation accounts if possible.
+     * 
+     * @see org.kuali.kfs.gl.batch.service.impl.PosterServiceImpl#postTransaction(
+     * org.kuali.kfs.gl.businessobject.Transaction, int, java.util.Map, org.kuali.kfs.gl.report.LedgerSummaryReport, java.io.PrintStream,
+     * org.kuali.kfs.sys.businessobject.UniversityDate, java.lang.String, java.io.PrintStream)
+     */
+    @Override
+    protected boolean postTransaction(Transaction tran, int mode, Map<String,Integer> reportSummary, LedgerSummaryReport ledgerSummaryReport,
```

Comment:
This is mostly the same as the superclass's version of the method, apart from a few noted additions and some cleanup related to generics usage.